src/use-positioner.ts

```diff
@@ -34,33 +36,30 @@ export function usePositioner(
       columnGutter
     )
   }
-  const [positioner, setPositioner] = React.useState<Positioner>(initPositioner)
-  const didMount = React.useRef(0)
 
-  // Create a new positioner when the dependencies change
-  useLayoutEffect(() => {
-    if (didMount.current) setPositioner(initPositioner())
-    didMount.current = 1
-    // eslint-disable-next-line
-  }, deps)
+  const [state, setState] = React.useState<UsePositionerState>(() => ({
+    deps,
+    options,
+    positioner: initPositioner(),
+  }))
+
+  let {deps: previousDeps, options: previousOptions, positioner} = state
 
-  // Updates the item positions any time a prop potentially affecting their
+  // Create a new positioner when the dependencies change.
+  // Also updates the item positions any time a prop potentially affecting their
   // size changes
-  useLayoutEffect(() => {
-    if (didMount.current) {
-      const cacheSize = positioner.size()
-      const nextPositioner = initPositioner()
-      let index = 0
-
-      for (; index < cacheSize; index++) {
-        const pos = positioner.get(index)
-        nextPositioner.set(index, pos !== void 0 ? pos.height : 0)
-      }
+  const hasDepsChanged = !areDepsEqual(deps, previousDeps)
+  const hasOptionsChanged = !areOptionsEqual(options, previousOptions)
+  if (hasDepsChanged || hasOptionsChanged) {
```

Comment:
The update in 1 pass ;-)

Also there will be no more 2 re-initializations. Both dependencies and options may be updated at the same time.